src/mono/wasm/runtime/binding_support.js

```diff
@@ -1551,24 +1556,23 @@ var BindingSupportLib = {
 				signature = Module.mono_method_get_call_signature (method);
 
 			return function() {
-				return BINDING.call_method (method, null, signature, arguments);
+				try {
+					var args = [...arguments];
+					if (args.length > 0 && Array.isArray (args[0]))
+						args[0] = BINDING.js_array_to_mono_array (args[0], true);
+
+					let result = BINDING.call_method (method, null, signature, args);
+					if (typeof (result.then) == "function")
+						return result;
+
+					return new Promise ((resolve, _) => resolve (result));
```

Comment:
This could be simplified to:

```suggestion
					return Promise.resolve(result);
```